Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+/*****************************************************************************
+ * VLCRendererDiscovererManager.swift
+ *
+ * Copyright © 2018 VLC authors and VideoLAN
+ * Copyright © 2018 Videolabs
+ *
+ * Authors: Soomin Lee <bubu@mikan.io>
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+class VLCRendererDiscovererManager: NSObject {
+    @objc static let sharedInstance = VLCRendererDiscovererManager(presentingViewController: nil)
+
+    // Array of RendererDiscoverers(Chromecast, UPnP, ...)
+    @objc dynamic var discoverers: [VLCRendererDiscoverer] = [VLCRendererDiscoverer]()
+
+    @objc lazy var actionSheet: VLCActionSheet = {
+        let actionSheet = VLCActionSheet()
+        actionSheet.delegate = self
+        actionSheet.dataSource = self
+        actionSheet.modalPresentationStyle = .custom
+        actionSheet.setAction { [weak self] (item) in
+            if let rendererItem = item as? VLCRendererItem {
+                self?.setRendererItem(rendererItem: rendererItem)
+            }
+        }
+        return actionSheet
+    }()
+
+    @objc var presentingViewController: UIViewController?
+
+    @objc var rendererButtons: [UIButton] = [UIButton]()
+
+    fileprivate init(presentingViewController: UIViewController?) {
+        self.presentingViewController = presentingViewController
+        super.init()
+    }
+
+    // Returns renderers of *all* discoverers
+    @objc func getAllRenderers() -> [VLCRendererItem] {
+        var renderers = [VLCRendererItem]()
+
+        for discoverer in discoverers {
+            renderers += discoverer.renderers
+        }
+        return renderers
+    }
+
+    fileprivate func isDuplicateDiscoverer(with description: VLCRendererDiscovererDescription) -> Bool {
+        for discoverer in discoverers {
+            if discoverer.name == description.name {
+                return true
+            }
+        }
+        return false
+    }
+
+    @discardableResult @objc func start() -> Bool {
+        // Gather potential renderer discoverers
+        guard let tmpDiscoverers: [VLCRendererDiscovererDescription] = VLCRendererDiscoverer.list() else {
+            return false
+        }
+        for discoverer in tmpDiscoverers {
+            if !isDuplicateDiscoverer(with: discoverer) {
+                if let rendererDiscoverer = VLCRendererDiscoverer(name: discoverer.name) {
+                    if rendererDiscoverer.start() {
+                        rendererDiscoverer.delegate = self
+                        discoverers.append(rendererDiscoverer)
+                    } else {
+                        print("Unable to start renderer discoverer with name: \(rendererDiscoverer.name)")
```

Comment:
don't use prints but errors or asserts. When does this case happen ? Is there something the programmer did wrong or does the user need feedback that this failed ?